assets/src/edit-story/components/canvas/eyedropperLayer.js

```diff
@@ -0,0 +1,259 @@
+
+      // Sometimes initial setting is not enough.
+      ctx.imageSmoothingEnabled = false;
+      ctx.mozImageSmoothingEnabled = false;
+      ctx.webkitImageSmoothingEnabled = false;
+      ctx.msImageSmoothingEnabled = false;
+
+      // Draw enlarged cropped image.
+      ctx.clearRect(0, 0, magnifierSize, magnifierSize);
+      ctx.drawImage(
+        imgRef.current,
+        Math.round(x - magnifierPixels / 2),
+        Math.round(y - magnifierPixels / 2),
+        magnifierPixels,
+        magnifierPixels,
+        0,
+        0,
+        magnifierSize,
+        magnifierSize
+      );
+
+      // Draw center rectangle for better aiming.
+      ctx.beginPath();
+      ctx.rect(
+        Math.round(magnifierSize / 2) - magnifierRectSize / 2,
+        Math.round(magnifierSize / 2) - magnifierRectSize / 2,
+        magnifierRectSize,
+        magnifierRectSize
+      );
+      ctx.stroke();
+    }
+  };
+
+  const onMouseMove = (e) => {
+    const { left, top, width, height } =
+      fullbleedContainer.getBoundingClientRect();
+    const x = (e.clientX - left) * (pageWidth / width);
+    const y = (e.clientY - top) * (fullHeight / height);
+
+    if (x < 0 || y < 0 || x > width || y > height) {
+      return;
+    }
+
+    // Move magnifier canvas.
+    magnifierInfo.current.style.transform = `translate(${x}px, ${y}px)`;
+
+    // Redraw magnifier canvas.
+    magnify(x, y);
+
+    // Get and print pixel color.
+    const rgbaObject = getColorFromPixelData(eyedropperPixelData, x, y, width);
+    const { r, g, b, a } = rgbaObject;
+    const hex = rgba(r, g, b, a);
+    magnifierColor.current.style.background = `rgba(${r},${g},${b},${a})`;
+    magnifierColor.current.style.color = readableColor(
+      hex,
+      '#333',
+      '#EDEDED',
+      false
+    );
+    magnifierColor.current.innerText = hex;
+  };
+
+  const onClick = (e) => {
+    const { left, top, width, height } =
+      fullbleedContainer.getBoundingClientRect();
+    const x = (e.clientX - left) * (pageWidth / width);
+    const y = (e.clientY - top) * (fullHeight / height);
+
+    const rgbaObject = getColorFromPixelData(
+      eyedropperPixelData,
+      x,
+      y,
+      pageWidth
+    );
+    eyedropperCallback(rgbaObject);
+  };
+
+  return (
+    <EyedropperBackground>
+      <DisplayPageArea withSafezone={false} showOverflow>
+        {/* eslint-disable-next-line styled-components-a11y/click-events-have-key-events, styled-components-a11y/no-static-element-interactions */}
+        <EyedropperCanvas
+          ref={eyedropperCanvas}
+          onClick={onClick}
+          onMouseMove={onMouseMove}
+        >
+          <img ref={imgRef} src={img} alt="" />
+          <Magnifier ref={magnifierInfo}>
+            <MagnifierCanvas
+              ref={magnifier}
+              width={magnifierSize}
+              height={magnifierSize}
+            />
+            <ColorInfo
+              style={{
+                top: magnifierSize + 20 + 10,
```

Comment:
What are the `20` and `10` for? Let's create constants instead of using magic numbers.